docs/book/overview.md

```diff
@@ -0,0 +1,43 @@
+# Overview
+
+vSphere CSI driver consists on 2 main components
+- vSphere CSI Controller
+- Syncer
+  - Metadata Syncer
+  - Full Sync
+  - CnsOperator
+
+### vSphere CSI Controller
+The vSphere Container Storage Interface (CSI) Driver provides a CSI interface used by Container Orchestrators to manage the lifecycle of vSphere volumes.
+
+CSI driver is responsible for volume provisioning, attaching and detaching the volume to VMs, mounting, formatting and unmounting volumes from the pod within the node VM etc. 
+
+### Syncer
+#### Metadata Syncer
+The syncer is responsible for pushing PV, PVC and Pod metadata to CNS. These metadata will be presented to vCenter or Storage Admin on vSphere UI
+
+#### Full Sync
+Full Sync is responsible keeping CNS upto date with all Kubernetes volume metadata PV, PVC and Pod metadata. Some cases where full sync would be helpful are listed below:
+
+- CNS goes down
+- CSI Pod goes down
+- API server goes down / K8s core services goes down
+- VC is restored to a backup point
+- etcd is restored to a backup point
+
+#### CnsOperator
+CnsOperator is used in the context of Project Pacific. CnsOperator is responsible for
+
+- attach/detach volumes to node VMs for Tanzu K8S Grid
+- pushing PV, PVC and Pod metadata to CNS from Tanzu K8S Grid
+
+## vSphere CSI driver supports various flavors
+- Vanilla Kubernetes
+  - vSphere CSI driver for block volumes
```

Comment:
We dont really have separate drivers. So, maybe we should not say this? Instead we should explain what do we mean by these k8s flavors.